mapper/orm/src/main/java/org/hibernate/search/mapper/orm/outbox/impl/OutboxTableSendingPlan.java

```diff
@@ -49,18 +50,43 @@ public void discard() {
 	@Override
 	public <R> CompletableFuture<MultiEntityOperationExecutionReport<R>> sendAndReport(
 			EntityReferenceFactory<R> entityReferenceFactory) {
+		if ( session.isOpen() ) {
+			return sendAndReportOnSession( session, entityReferenceFactory );
+		}
+
+		// See https://hibernate.atlassian.net/browse/HSEARCH-4198
+		// When JTA is enabled with Spring, the shouldReleaseBeforeCompletion strategy is used by default.
+		// Solution inspired by org.hibernate.envers.internal.synchronization.AuditProcess#doBeforeTransactionCompletion.
+		Session temporarySession = null;
+		try {
+			temporarySession = session.sessionWithOptions()
+					.connection()
+					.autoClose( false )
+					.connectionHandlingMode( PhysicalConnectionHandlingMode.DELAYED_ACQUISITION_AND_RELEASE_AFTER_TRANSACTION )
+					.openSession();
+			return sendAndReportOnSession( temporarySession, entityReferenceFactory );
+		}
+		finally {
+			if ( temporarySession != null ) {
+				temporarySession.close();
+			}
+		}
```

Comment:
A `try-with-resources` block would be nice here. And that's not just nitpicking:

* with try-with-resources, if `sendAndReportOnSession` throws an exception and `temporarySession.close()` also throws an exception, the second exception will correctly be added as suppressed to the first one.
* with `try/finally`, in the same situation, the first exception will be ignored and only the second one (from `temporarySession.close()`) will be propagated.